pkg/collector/corechecks/snmp/internal/profile/profile.go

```diff
@@ -0,0 +1,84 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package profile contains profile related code
+package profile
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/networkdevice/profile/profiledefinition"
+)
+
+// GetProfiles returns profiles depending on various sources:
+//   - init config profiles
+//   - yaml profiles
+//   - downloaded json gzip profiles
+//   - remote config profiles
```

Comment:
Currently, the init config profiles take precedence over all profiles (also current implementation with this PR), however I'd expect this behaviour to be different with remote config profiles or downloaded profiles.
I f I go through the Profile generator (with or without RC direct changes), I'd expect these profiles to have precedence over the "old ones" in the config.
Did we have this discussion over new priorities ?

Reply:
@NouhaManai96 

Indeed, in most cases, we likely want the new ways (profiles from Profile Editor, manual or RC) to have precedence.

But I think for config from init_config or instance config, they should take precedence anyway. My rational is that:

1/ configs that are more specific should take precedence, to make it possible to override higher level configs.
There is only one profile.json.gz (less specific), but there are many conf.yaml, conf1.yaml, conf2.yaml in snmp.d (more specific). User can decide that conf2.yaml will have a differente behaviour by using init_config profile definition.

2/ related to 1/, if init_config have lower precedence, user won't be able to use init_config profiles/metrics to just override in conf2.yaml.

Reply:
I think the precedence should be:

1/ init_config profiles
2/ RC profiles (Profile Editor)
3/ downloaded profiles (Profile Editor)
4/ Yaml profiles

I can open a thread to discuss about it.